From 83d8e64e3d067624125587ddf14d115e1c787cfa Mon Sep 17 00:00:00 2001
From: Lily Zhang <r58066@freescale.com>
Date: Mon, 11 Oct 2010 18:29:12 +0800
Subject: [PATCH] ENGR00132430-3 add MT29F16G08ABACA NAND flash support

Add MT29F16G08ABACA NAND flash support. Although the actual
oob size of MT29F16G08ABACA NAND is 224 byte, set it as 218
byte since MX53 NFC controller mentions "In case ECC is
enabled, the spare area size must be less or equal 218B"

Signed-off-by: Lily Zhang <r58066@freescale.com>
---
 drivers/mtd/nand/nand_device_info.c |   23 +++++++++++++++++++++++
 drivers/mtd/nand/nand_ids.c         |    1 +
 2 files changed, 24 insertions(+), 0 deletions(-)

diff --git a/drivers/mtd/nand/nand_device_info.c b/drivers/mtd/nand/nand_device_info.c
index 044e397..cb8a4dc 100644
--- a/drivers/mtd/nand/nand_device_info.c
+++ b/drivers/mtd/nand/nand_device_info.c
@@ -495,6 +495,29 @@ static struct nand_device_info nand_device_info_table_type_2[] __initdata = {
 	.tRHOH_in_ns              = -1,
 	NULL,
 	},
+	{
+	.end_of_table             = false,
+	.manufacturer_code        = 0x2c,
+	.device_code              = 0x48,
+	.cell_technology          = NAND_DEVICE_CELL_TECH_SLC,
+	.chip_size_in_bytes       = 2LL*SZ_1G,
+	.block_size_in_pages      = 128,
+	/* TODO: The actual oob size for MT29F16G08ABACA is
+	224 bytes. Use oob 218 bytes since MX53 NFC controller
+	mentions the spare area size must be less or equal 218
+	byte if ECC is enabled */
+	.page_total_size_in_bytes = 4*SZ_1K + 218,
+	.ecc_strength_in_bits     = 8,
+	.ecc_size_in_bytes        = 512,
+	.data_setup_in_ns         = 15,
+	.data_hold_in_ns          = 10,
+	.address_setup_in_ns      = 20,
+	.gpmi_sample_delay_in_ns  = 6,
+	.tREA_in_ns               = 20,
+	.tRLOH_in_ns              = 5,
+	.tRHOH_in_ns              = 15,
+	"MT29F16G08ABACA(2GB)",
+	},
 	{true}
 };
 
diff --git a/drivers/mtd/nand/nand_ids.c b/drivers/mtd/nand/nand_ids.c
index 77a100e..922fa4e 100644
--- a/drivers/mtd/nand/nand_ids.c
+++ b/drivers/mtd/nand/nand_ids.c
@@ -109,6 +109,7 @@ struct nand_flash_dev nand_flash_ids[] = {
 	{"NAND 2GiB 3,3V 8-bit",	0xD5, 0, 2048, 0, LP_OPTIONS},
 	{"NAND 2GiB 1,8V 16-bit",	0xB5, 0, 2048, 0, LP_OPTIONS16},
 	{"NAND 2GiB 3,3V 16-bit",	0xC5, 0, 2048, 0, LP_OPTIONS16},
+	{"NAND 2GiB 3,3V 8-bit",	0x48, 0, 2048, 0, LP_OPTIONS},
 
 	/* 32 Gigabit ,only use 2G due to the linux mtd limitation*/
 	{"NAND 4GiB 3,3V 8-bit",	0xD7, 0, 2048, 0, LP_OPTIONS},
-- 
1.5.4.4

